android/src/main/java/com/swmansion/rnscreens/Screen.kt

```diff
@@ -246,6 +248,40 @@ class Screen(context: ReactContext?) : FabricEnabledViewGroup(context) {
 
     var nativeBackButtonDismissalEnabled: Boolean = true
 
+    fun startRemovalTransition() {
+        if (!isBeingRemoved) {
+            isBeingRemoved = true
+            startTransitionRecursive(this)
+        }
+    }
+
+    private fun startTransitionRecursive(parent: ViewGroup?) {
+        parent?.let {
+            for (i in 0 until it.childCount) {
+                val child = it.getChildAt(i)
+                if (child.javaClass.simpleName.equals("CircleImageView")) {
+                    // SwipeRefreshLayout class which has CircleImageView as a child,
+                    // does not handle `startViewTransition` properly.
+                    // It has a custom `getChildDrawingOrder` method which returns
+                    // wrong index if we called `startViewTransition` on the views on new arch.
+                    // We add a simple View to make bump the number of children so it works.
+                    // TODO: find a better way to handle this scenario
+                    it.addView(View(context), i)
```

Comment:
This is wild 🙈

Reply:
Although it is something similar @maciekstosio is handling with the not respected zIndex during pop transition in custom header.